### ISSUE TITLE
Bookmark are not syncing between iOS and Android

### ISSUE THREAD

**Issue by SimonBasca** [mozilla-mobile/firefox-ios#8792]
## Preconditions:
- Logged in to FxA on IOS and Android
## Steps to reproduce
1. Go to iOS device
2. Open and bookmark a page
3. Sync on iOS 
4. Sync on Android
### Expected behavior
- The bookmark should sync across devices
### Actual behavior
- The bookmark is not synced after syncing on iOS and Android. After logging out and logging in to FxA on Android the bookmark in shown in the bookmark panel. Also note that the bookmark is synced across iOS devices. 
### Device & build information
* Device: iPhone Xr (14.5), Galaxy A71 (11)
* Build version: Firefox iOS main 3b9c2d5 and latest Fenix 91.a01.

### Note:
- Bookmarks are syncing from Android to iOS but not the other way around.

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/FXIOS-2924)

**Comment by dnarcese** [comment]
@lougeniaC64 Do you know why this would be happening and what we would need to fix?

**Comment by lougeniaC64** [comment]
> @lougeniaC64 Do you know why this would be happening and what we would need to fix?

@dnarcese I'm not sure offhand but while attempting to replicate this issue I could only do so on Firefox Beta (90.0.0-beta 6), not Release or Nightly. I know that the android components team has done a lot of work on the places rust component but I'll raise this issue in the SACI slack channel for team visibility.

**Comment by mhammond** [comment]
I'd suggest adding a desktop device to the constellation - this will probably help to determine which of the platforms is the problem - we'd expect to see 2 of the 3 with a consistent set, making the odd-one-out the problem child. Once we've identified the problem child, we probably want logs from it - which, if it is android, means via adb or an emulator. I'm not sure how you get logs from iOS, but @lougeniaC64 or @skhamis might.

**Comment by isabelrios** [comment]
In case this helps, we have sync tests running between iOS <-> Desktop and they are fine syncing bookmarks.
See https://mozilla.slack.com/archives/CAFC45W5A/p1628758371012100

**Comment by lougeniaC64** [comment]
@mhammond Thanks for the suggestion. I added desktop release to the party and was able to see bookmarks I'd created on my iPhone on Desktop. I tried to reproduce this issue again with Firefox Beta for Android and wasn't able to. Turns out my Beta auto-updated to 91 (specifically 91.0.0-beta 6) while I wasn't looking which leads me to believe there was an issue pushed in 90 that was resolved in 91.
